src/p_saveg.c

```diff
@@ -257,14 +257,17 @@ static void saveg_write_mapthing_t(mapthing_t *str)
 
 static void saveg_read_thinker_t(thinker_t *str)
 {
+    void *func;
+
     // struct thinker_s* prev;
     str->prev = saveg_readp();
 
     // struct thinker_s* next;
     str->next = saveg_readp();
 
     // think_t function;
-    str->function.v = (actionf_v)(intptr_t)saveg_readp();
+    func = saveg_readp();
+    str->function.v = (actionf_v)(intptr_t)func;
```

Comment:
All of these addresses don't work anyway. Perhaps we should redo this for clarity:
```c
saveg_readp();
str->function.v = NULL;
```

Reply:
Yes. I wasn't sure and wanted to get rid of the warning in the first place, but this is even better.